### PR TITLE
NMS-8763: Documenting users, groups and security roles

### PR DESCRIPTION
**Pull request by indigo423**
Create documentation how to change user passwords. Additionally create documentation how to create and modify users, groups, security roles and on-call roles. Merge with existing documentation about  notification users.

Cleanups:
* pom files for guide-concepts and guide-docs which don't exist anymore
* Normalized the product name to be _italic_

Issue Tracker:
* JIRA: http://issues.opennms.org/browse/NMS-8763

### CODE REVIEW

**Comment by indigo423** [comment]
@RangerRick what is required to get the Bamboo checks enabled for this pull request?

**Comment by RangerRick** [comment]
@indigo423 it should auto-build as long as it's in the main OpenNMS repository, unless it's an old branch you've resurrected.

File: opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+.Functions and existing system roles in _{opennms-product-name}_
+[options="header, autowidth"]
+|===
+| Security Role Name  | Description
+| _anyone_            | In case the `opennms-webapp-remoting` package is installed, any user can download the _Java Webstart_ installation package for the remote poller from http://opennms.server:8980/opennms-remoting/webstart/app.jnlp.
+| _ROLE_ANONYMOUS_    | Allows _HTTP OPTIONS_ request to show allowed HTTP methods on a ReST resources and the login and logout page of the Web User Interface.
+| _ROLE_ADMIN_        | Permissions to create, read, update and delete in the Web User Interface and the _ReST API_.
+| _ROLE_ASSET_EDITOR_ | Permissions to just update the asset records from nodes.
+| _ROLE_DASHBOARD_    | Allow users to just have access to the _Dashboard_.
+| _ROLE_JMX_          | ?
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc]: See https://github.com/OpenNMS/opennms/blame/develop/opennms-doc/guide-admin/src/asciidoc/text/rmi.adoc#L29

File: opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+| _ROLE_MINION_       | ?
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc]: See https://github.com/OpenNMS/opennms/blame/develop/opennms-doc/guide-install/src/asciidoc/text/minion/configure.adoc#L13

**Comment by indigo423** [comment]
@RangerRick thanks for resurrecting the old branch.
@j-white thanks for hints, added the missing documentation parts

File: opennms-doc/guide-admin/src/asciidoc/text/user-management/users.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+_Users_ are entities with login accounts in the _{opennms-product-name}_ system.
+Ideally each user corresponds to a person.
+An _{opennms-product-name}_ user represents an actor which may be granted permissions in the system by associating _Security Roles_.
+_User_ can be created in _{opennms-product-name}_ using it's embedded file based storage for users with login name and password and are managed through the Web User Interface.
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/user-management/users.adoc]: I find this sentence hard to follow.

File: opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[ga-role-user-management-roles]]
+=== Security Roles
+
+A _Security Roles_ is a set of permissions and can be assigned to an _User_.
+They regulate access to the Web User Interface and the _ReST API_ to exchange monitoring and inventory information.
+In case of a distributed installation, the _MINION_ or _Remote Poller_ instances interact with _{opennms-product-name}_ and require specific permissions which are defined in the _Security Roles_ _ROLE_MINION_ and _ROLE_REMOTING_.
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc]: MINION vs Minion

File: opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+.Functions and existing system roles in _{opennms-product-name}_
+[options="header, autowidth"]
+|===
+| Security Role Name  | Description
+| _anyone_            | In case the `opennms-webapp-remoting` package is installed, any user can download the _Java Webstart_ installation package for the remote poller from http://opennms.server:8980/opennms-remoting/webstart/app.jnlp.
+| _ROLE_ANONYMOUS_    | Allows _HTTP OPTIONS_ request to show allowed HTTP methods on a ReST resources and the login and logout page of the Web User Interface.
+| _ROLE_ADMIN_        | Permissions to create, read, update and delete in the Web User Interface and the _ReST API_.
+| _ROLE_ASSET_EDITOR_ | Permissions to just update the asset records from nodes.
+| _ROLE_DASHBOARD_    | Allow users to just have access to the _Dashboard_.
+| _ROLE_JMX_          | Allows retrieving JMX metrics but does not allow executing MBeans, even if they just return simple values.
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/user-management/security-roles.adoc]: Should we elaborate that this is for the OpenNMS JVM?

File: opennms-doc/guide-admin/src/asciidoc/text/user-management/users.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+_Users_ are entities with login accounts in the _{opennms-product-name}_ system.
+Ideally each user corresponds to a person.
+An _{opennms-product-name}_ user represents an actor which may be granted permissions in the system by associating _Security Roles_.
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/user-management/users.adoc]: user vs _User_

**Comment by indigo423** [comment]
@j-white thank you for review. Improved the parts

**Comment by j-white** [comment]
Thanks for the updates @indigo423. +1 from me
